### PR TITLE
Federation apiserver batch job client

### PR DESCRIPTION
**Pull request by jianhuiz**
generate internal and 1.5 client for federated job
requires #35943

@quinton-hoole @nikhiljindal @deepak-vij
#34261

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35945)

### CODE REVIEW

**Comment by madhusudancs** [comment]
@k8s-bot ok to test

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@jianhuiz haven't reviewed yet, but you will also have to sign the Linux Foundation CLA for this PR to be merged.

**Comment by quinton-hoole** [comment]
Thanks @jianhuiz !  Do you have the controller code for this yet?  If not it's going to be tough to try to get it into v1.5 on time - the code freeze is Nov 8, one week from today.  FYI.

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -59,8 +59,8 @@ ${clientgen} "$@"
...
+${clientgen} --clientset-name=federation_internalclientset --clientset-path=k8s.io/kubernetes/federation/client/clientset_generated --input="../../federation/apis/federation/","api/","extensions/","batch/" --included-types-overrides="api/Service,api/Namespace,extensions/ReplicaSet,api/Secret,extensions/Ingress,extensions/Deployment,extensions/DaemonSet,api/ConfigMap,api/Event,batch/Job"   "$@"
+${clientgen} --clientset-name=federation_release_1_5 --clientset-path=k8s.io/kubernetes/federation/client/clientset_generated --input="../../federation/apis/federation/v1beta1","api/v1","extensions/v1beta1","batch/v1" --included-types-overrides="api/v1/Service,api/v1/Namespace,extensions/v1beta1/ReplicaSet,api/v1/Secret,extensions/v1beta1/Ingress,extensions/v1beta1/Deployment,extensions/v1beta1/DaemonSet,api/v1/ConfigMap,api/v1/Event,batch/Job"   "$@"
```
Comment (nikhiljindal) [inline hack/update-codegen.sh]: Shouldnt it be batch/v1/Job at the end?

Comment (jianhuiz) [inline hack/update-codegen.sh]: you are right. updated

**Comment by nikhiljindal** [comment]
Sorry @jianhuiz This missed 1.5.
Lets get this in 1.6. Can you update the PR to update 1.6 clientset instead of 1.5 one?

**Comment by nikhiljindal** [comment]
Any update @jianhuiz ?

**Comment by jianhuiz** [comment]
@nikhiljindal I thought this has to be after 1.5 release, all clientset reference moved to 1.6. I was waiting the current 1.5 clientset removed and 1.6 clientset generated.

**Comment by nikhiljindal** [comment]
Yes someone has to do that - create a 1.6 clientset and move everything to use that.

**Comment by jianhuiz** [comment]
I can do the federation 1.6 clientset if it's the right time and has not been taken

**Comment by nikhiljindal** [comment]
Yes sure. Go ahead. Cc @caesarxuchao as fyi

**Comment by caesarxuchao** [comment]
I sent https://github.com/kubernetes/kubernetes/pull/38154 to rename "release_1_5" clientsets to just "clientset". The reason is we now have client-go to track different versions of clientsets.
@nikhiljindal Federation clientsets are not tracked in client-go. Do people really need different releases of clientset for federation? Can they checkout release tags of kubernetes if they want to use a specific version of the federationclientsets? If the answers are yes, I will rename "federation_release_1_5" to "federation_clientset" in #38154.

**Comment by nikhiljindal** [comment]
For federation code in repo we use the same Client version everywhere.
For out of repo use, can they checkout release tag using godeps? If yes then renaming sounds good.

**Comment by caesarxuchao** [comment]
> For out of repo use, can they checkout release tag using godeps?

Yes. Thanks for the confirmation.

**Comment by jianhuiz** [comment]
I'll update this PR after #38154 merged

**Comment by jianhuiz** [comment]
rebased and regenerated for federation_clientset

**Comment by nikhiljindal** [comment]
Thanks, LGTM

**Pull request merged**